# 가격 변화가 매출에 미치는 영향 분석

In [1]:
import warnings
warnings.filterwarnings('ignore')
import matplotlib.font_manager as fm
for i in fm.fontManager.ttflist:
    if i.name.find('Gulim') != -1:
        print(i.name)

Gulim
New Gulim


## 변수 탐색

In [3]:
# 데이터 불러오기
import pandas as pd

df = pd.read_csv("c:/githome/hipython_rep/data/walmart.csv")

# 기본 정보 확인
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 550068 entries, 0 to 550067
Data columns (total 10 columns):
 #   Column                      Non-Null Count   Dtype 
---  ------                      --------------   ----- 
 0   User_ID                     550068 non-null  int64 
 1   Product_ID                  550068 non-null  object
 2   Gender                      550068 non-null  object
 3   Age                         550068 non-null  object
 4   Occupation                  550068 non-null  int64 
 5   City_Category               550068 non-null  object
 6   Stay_In_Current_City_Years  550068 non-null  object
 7   Marital_Status              550068 non-null  int64 
 8   Product_Category            550068 non-null  int64 
 9   Purchase                    550068 non-null  int64 
dtypes: int64(5), object(5)
memory usage: 42.0+ MB


- 총 데이터 수는 550068개이고, 10개의 column이 있으며 결측값은 없다.
- column의 종류는 범주형 변수 5개와 수치형 변수 5개로 이루어져 있다. 

In [8]:
df.isnull().sum()

User_ID                       0
Product_ID                    0
Gender                        0
Age                           0
Occupation                    0
City_Category                 0
Stay_In_Current_City_Years    0
Marital_Status                0
Product_Category              0
Purchase                      0
dtype: int64

모든 column에 결측값은 다 0이다.

In [5]:
#데이터 샘플 보기
df.head()

User_ID Product_ID Gender   Age  Occupation City_Category  \
0  1000001  P00069042      F  0-17          10             A   
1  1000001  P00248942      F  0-17          10             A   
2  1000001  P00087842      F  0-17          10             A   
3  1000001  P00085442      F  0-17          10             A   
4  1000002  P00285442      M   55+          16             C   

  Stay_In_Current_City_Years  Marital_Status  Product_Category  Purchase  
0                          2               0                 3      8370  
1                          2               0                 1     15200  
2                          2               0                12      1422  
3                          2               0                12      1057  
4                         4+               0                 8      7969

In [6]:
df.describe()

User_ID     Occupation  Marital_Status  Product_Category  \
count  5.500680e+05  550068.000000   550068.000000     550068.000000   
mean   1.003029e+06       8.076707        0.409653          5.404270   
std    1.727592e+03       6.522660        0.491770          3.936211   
min    1.000001e+06       0.000000        0.000000          1.000000   
25%    1.001516e+06       2.000000        0.000000          1.000000   
50%    1.003077e+06       7.000000        0.000000          5.000000   
75%    1.004478e+06      14.000000        1.000000          8.000000   
max    1.006040e+06      20.000000        1.000000         20.000000   

            Purchase  
count  550068.000000  
mean     9263.968713  
std      5023.065394  
min        12.000000  
25%      5823.000000  
50%      8047.000000  
75%     12054.000000  
max     23961.000000

In [ ]:
#'User_ID'의 값 목록 보기
df['User_ID'].value_counts()

User_ID
1001680    1026
1004277     979
1001941     898
1001181     862
1000889     823
           ... 
1004991       7
1005391       7
1000094       7
1005608       7
1000708       6
Name: count, Length: 5891, dtype: int64

- User_ID의 목록을 보면 '1001680' 고객님의 구매 횟수가 1026번으로 가장 많다.
- 최소 구매 횟수 고객님의 구매 수가 6번으로 walmart에는 한 번만 구매하는 고객님은 없음을 알 수 있다.

In [11]:
df['User_ID'].value_counts().describe()

count    5891.000000
mean       93.374300
std       107.190049
min         6.000000
25%        26.000000
50%        54.000000
75%       117.000000
max      1026.000000
Name: count, dtype: float64

- 고객님들의 구매 횟수의 평균이 93.374이고 표준편차는 107.190이다
- 75%, 즉 구매 횟수가 117번 이상인 고객님부터는 VIP 고객님으로 간주할 수 있다.

In [12]:
df['Purchase'].describe()

count    550068.000000
mean       9263.968713
std        5023.065394
min          12.000000
25%        5823.000000
50%        8047.000000
75%       12054.000000
max       23961.000000
Name: Purchase, dtype: float64

- 구매 가격의 평균값은 9263달러, 최댓값은 23961달러, 최솟값은 12달러로 제품간 가격 편차는 큰 편이다.

# 데이터 전처리

### 결측값은 없으므로 결측치 처리를 하지 않는 대신 이상치를 처리하도록 한다.
- 이상치(Outlier) 보기 (Age, Occupation, Stay_In_Current_City_Years, Marital_Status, Product_Category)

In [13]:
#폰트 지정
from matplotlib import rcParams
rcParams['font.family'] = 'Gulim'
rcParams['axes.unicode_minus'] = False

In [ ]:
import matplotlib.pyplot as plt

fig = plt.figure(figsize=(12,4))
axes = fig.add_subplot(1,1,1)

axes.boxplot(
             x=[
                 df['pct_carrier_delay'],
                 df['pct_atc_delay'],
                 df['pct_weather_delay']
             ],
             labels=['pct_carrier_delay', 'pct_atc_delay', 'pct_weather_delay'])
plt.show()